webapp/src/components/BuyParcelByMortgagePage/MortgageForm/MortgageForm.js

```diff
@@ -0,0 +1,324 @@
+    }
+
+    if (
+      interestRate < MINIMUM_INTEREST_RATE ||
+      punitoryRate < MINIMUM_INTEREST_RATE
+    ) {
+      formErrors.push(
+        t('mortgage.errors.minimum_interest_rate', {
+          minimum: MINIMUM_INTEREST_RATE
+        })
+      )
+    }
+
+    if (
+      interestRate > MAXIMUM_INTEREST_RATE ||
+      punitoryRate > MAXIMUM_INTEREST_RATE
+    ) {
+      formErrors.push(
+        t('mortgage.errors.maximum_interest_rate', {
+          maximum: MAXIMUM_INTEREST_RATE
+        })
+      )
+    }
+
+    if (payableAt > duration) {
+      formErrors.push(t('mortgage.errors.duration_gte_payable'))
+    }
+
+    if (differenceInDays(expiresAt, today) > MAXIMUM_DAY_INTERVAL) {
+      formErrors.push(
+        t('parcel_publish.errors.maximum_expiration', {
+          date: this.formatFutureDate(MAXIMUM_DAY_INTERVAL)
+        })
+      )
+    }
+
+    if (differenceInDays(expiresAt, today) < MINIMUM_DAY_INTERVAL) {
+      formErrors.push(
+        t('parcel_publish.errors.minimum_expiration', {
+          date: this.formatFutureDate(MINIMUM_DAY_INTERVAL + 1)
+        })
+      )
+    }
+
+    if (formErrors.length === 0) {
+      onPublish({
+        duration: new Date(duration).getTime(),
+        payableAt: new Date(payableAt).getTime(),
+        expiresAt: new Date(expiresAt).getTime(),
+        amount: Number(amount),
+        interestRate,
+        punitoryRate,
+        parcel
+      })
+    } else {
+      this.setState({ formErrors })
+    }
+  }
+
+  formatFutureDate(addedDays, date = new Date()) {
+    date = addDays(date, addedDays)
+    return formatDate(date, INPUT_FORMAT)
+  }
+
+  render() {
+    const { publication, isTxIdle, isDisabled, onCancel } = this.props
+    const {
+      amount,
+      payableAt,
+      expiresAt,
+      interestRate,
+      duration,
+      punitoryRate,
+      formErrors
+    } = this.state
+
+    const isPending =
+      publication.tx_status === txUtils.TRANSACTION_STATUS.pending
+
+    return (
+      <Form
+        className="MortgageForm"
+        onSubmit={preventDefault(this.handlePublish)}
+        error={!!formErrors}
+      >
+        <Form.Field>
+          <label>{t('mortgage.oracle')}</label>
+          <p className="address-input">{'MANA <> BANCOR'}</p>
+          <AddressBlock
+            hasLink={false}
+            isUser={false}
+            scale={6}
+            address={getKyberOracleAddress()}
+            hasTooltip={false}
+          />
+        </Form.Field>
+        <Grid divided={'vertically'} className="MortgageFormGrid">
+          <Grid.Row columns={3}>
+            <Grid.Column mobile={16} computer={5}>
+              <Form.Field>
+                <label>{t('mortgage.amount')}</label>
+                <Input
+                  type="number"
+                  placeholder={t('mortgage.amount_placeholder')}
+                  value={amount}
+                  required={true}
+                  onChange={e => this.handleChangeNumber(e, 'amount')}
```

Comment:
I like this, but I'm unsure if it's worth it performance wise. @cazala what do you think?

Reply:
I did it because I think it is better than having multiple handlers doing the same for each input